asdf/tags/core/tests/test_history.py

```diff
@@ -201,20 +201,22 @@ def test_strict_extension_check():
 
 
 def test_metadata_with_custom_extension(tmpdir):
-    class FractionType(types.CustomType):
-        name = "fraction"
-        organization = "nowhere.org"
-        version = (1, 0, 0)
-        standard = "custom"
-        types = [fractions.Fraction]
-
-        @classmethod
-        def to_tree(cls, node, ctx):
-            return [node.numerator, node.denominator]
-
-        @classmethod
-        def from_tree(cls, tree, ctx):
-            return fractions.Fraction(tree[0], tree[1])
+    with pytest.warns(AsdfDeprecationWarning, match=".*subclasses the deprecated CustomType.*"):
+
+        class FractionType(types.CustomType):
```

Comment:
Perhaps. I would prefer to limit the amount of refactoring that is done to our tests to limit the potential bugs introduced. If refactoring the tests to all subclass `CustomTestType` results in overall fewer changes than what is already in this PR than I would likely consider it a better option.

Reply:
This would eliminate all the `pytest.warns` contexts from every test file. The only place that would be left is in `asdf.tests.objects` (or `asdf.tests.__init__` if it moves back there).